### PR TITLE
Remove unused buildspec and add cc test reporting from GH actions

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves #NUMBER

**Overall change:**
- Big Oops!
- Run. Code Climate test coverage and reporting

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: buildspecs/buildspec-ci-branch.yml [hunk 1]
```diff
@@ -1,33 +0,0 @@
...
-      - npx apache2-license-checker
-      # Trigger Chromatic PR checks
-      - npx chromatic test run --build-script-name build:storybook --exit-once-uploaded --no-interactive
-      # Set up Code Climate PR checks
-      - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
-      - chmod +x ./cc-test-reporter
-      - ./cc-test-reporter before-build
-      - npm run test:local
-      # Submit Code Climate results
-      - ./cc-test-reporter after-build -t lcov --debug --exit-code 0
```
Comment (karinathomasbbc) [inline buildspecs/buildspec-ci-branch.yml:32]: Do we need to replicate these lines in the GH action

File: .github/workflows/simorgh-misc-checks.yml [hunk 2]
```diff
@@ -31,6 +31,12 @@ jobs:
       - name: Chromatic UI Tests
         run: npx chromatic test run --build-script-name build:storybook --exit-once-uploaded --no-interactive --project-token "$CHROMATIC_APP_CODE"
 
+      - name: Code Climate Test Coverage
+        run: |
+          curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter	
+          chmod +x ./cc-test-reporter
+          ./cc-test-reporter after-build -t lcov --debug --exit-code 0
+
```
Comment (karinathomasbbc) [inline .github/workflows/simorgh-misc-checks.yml]: Does this check maybe belong with the unit tests - I noticed in the old script, it ran `npm run test:local` - we would rather avoid doing this again unless absolutely necessary

Comment (chris-hinds) [reply .github/workflows/simorgh-misc-checks.yml]: good spot, giving that a go now.

**Review by karinathomasbbc — APPROVED** [review]
Nice 🐺

File: .github/workflows/simorgh-unit-tests.yml [hunk 3]
```diff
@@ -28,5 +29,15 @@ jobs:
         run: |
           npm ci
           npm run build
+
```
Comment (karinathomasbbc) [inline .github/workflows/simorgh-unit-tests.yml:32]: Is it worth updating the GH Action name?
```
name: Simorgh CI - Unit Tests & Code Coverage
```

**Review by jroebu14 — APPROVED** [review]

**Pull request merged**